### PR TITLE
Waffles Part 1: Import config waffles to django waffles 🧇

### PR DESCRIPTION
**Pull request by robhudson**
## Multi-line summary

Part 1 (this PR):
- Adds django-waffle library
- Adds management command to load current `ConfigValue`s into waffle tables
- Removes funnelcake code

Part 2 will:
- Remove old www_config import
- Update the template tag helper to use django-waffle

We need to roll this out in two deployments since we can't ensure the database will be updated when the template tag switch happens.
- Deployment 1: Add waffle table and import waffle switches
- Deployment 2: Switch template tags to use django-waffle

## Significant changes and points to review

Ensure waffle switches still work. We haven't added django-waffle's templatetag or anything here so all the www_config stuff should work as before.

## Issue / Bugzilla link

#15157

## Testing

After updating Python dependencies with:
```bash
make install-local-python-deps
```

Check that you currently have no waffle switches:
```bash
./manage.py waffle_switch -l
```

Import the switches into waffle:
```bash
./manage.py import_waffle_switches
```

Verify that you have the expected waffle switches:
```bash
./manage.py waffle_switch -l
```

### CODE REVIEW

**Review by alexgibson — COMMENTED** [review]
I tried following the instructions here but I get an error saying `ModuleNotFoundError: No module named 'waffle'`

File: bedrock/base/management/commands/import_waffle_switches.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+@alert_sentry_on_exception
+class Command(BaseCommand):
+    def handle(self, *args, **options):
+        prefix = "SWITCH_"
+
+        for config in ConfigValue.objects.all():
+            # Ignore funnelcakes and other yummy things.
+            if not config.name.startswith(prefix):
+                continue
```
Comment (alexgibson) [inline bedrock/base/management/commands/import_waffle_switches.py:21]: Remove mention of funnalcakes here?

**Review by alexgibson — COMMENTED** [review]
This worked well for me (sorry it took a day or so to get back to this) r+

One question: how does this effect switches that are not set in www-config, but in deployments / secrets?

**Comment by robhudson** [comment]
> One question: how does this effect switches that are not set in www-config, but in deployments / secrets?

Do you have an example of a switch being configured in this way?

**Comment by janbrasna** [comment]
> > how does this effect switches that are not set in www-config, but in deployments / secrets?
> 
> Do you have an example of a switch being configured in this way?

I believe e.g. the Cloud Run demos have set it directly in the .env — so the question perhaps is if anything set on the env will continue working, or only the db switches will be honored…(?)

**Comment by alexgibson** [comment]
> I believe e.g. the Cloud Run demos have set it directly in the .env — so the question perhaps is if anything set on the env will continue working, or only the db switches will be honored…(?)

In addition I was wondering if dev/stage/prod deployments might also have certain switches set directly environment variables, but I don't have any visibility there to check.

**Comment by stevejalim** [comment]
You can check the env vars set in the various `*values.yaml` files here https://github.com/mozilla-it/webservices-infra/tree/main/bedrock/k8s/bedrock @alexgibson - the only reference to any switch i can see is  `switch_newsletter_maintenance_mode: 'False'` which would become `SWITCH_NEWSLETTER_MAINTENANCE_MODE` as an env var

**Comment by alexgibson** [comment]
> the only reference to any switch I can see is switch_newsletter_maintenance_mode: 'False' which would become SWITCH_NEWSLETTER_MAINTENANCE_MODE as an env var

OK, so I guess we'd need to move this into the DB?

**Comment by stevejalim** [comment]
> OK, so I guess we'd need to move this into the DB?

I'd imagine it can wait till after this rollout. It might well be redundant, as I can't see it being used in Bedrock or `basket-client`

**Review by stevejalim — APPROVED** [review]
Haven't test-driven because Alex aleady has. Code LGTM.

Park of me is asking "Are we 100% confident funnelcake builds won't come back?" and am sure the answer is "No", but I'd love to know more about that

**Comment by stevejalim** [comment]
@robhudson If you want the waffle flags to be included in the DB export, you'll need to add them to this list https://github.com/mozilla/bedrock/blob/main/bin/export-db-to-sqlite.sh#L132-L176 though it sounds like you've already done this

The only reason _not_ to include the Switches in the DB would be if they are secret (eg they leak info via their name, for instance), because the DB is downloadable (deliberately) and contains no sensitive info.

**Pull request merged**